tests/phpunit/tests/html-api/wpHtmlSupportRequiredActiveFormatReconstruction.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Unit tests for the HTML API ensuring proper handling of behaviors related to
+ * active format reconstruction.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ *
+ * @since 6.5.0
+ *
+ * @group html-api
+ *
+ * @coversDefaultClass WP_HTML_Processor
+ */
+class Tests_HtmlApi_WpHtmlSupportRequiredActiveFormatReconstruction extends WP_UnitTestCase {
+	/**
+	 * Ensures that active formats are properly reconstructed when visiting text nodes,
+	 * verifying that the proper breadcrumbs are maintained when scanning through HTML.
+	 *
+	 * @ticket {TICKET_NUMBER}
+	 *
+	 * @since 6.5.0
+	 */
+	public function test_reconstructs_active_formats_on_text_nodes() {
+		$processor = WP_HTML_Processor::create_fragment( '<p><b>One<span><p>Two' );
+
+		$processor->next_tag( 'span' );
+		$this->assertSame(
+			array( 'HTML', 'BODY', 'P', 'B', 'SPAN' ),
+			$processor->get_breadcrumbs(),
+			'Should have identified the stack of open elements for the first text node.'
+		);
+
+		$this->assertTrue(
+			$processor->next_tag( 'p' ),
+			'Should have found second P element.'
+		);
+
+		/*
+		 * There are two ways this test could fail. One is to appropriately find the
+		 * second text node but fail to reconstruct the implicitly-closed B element.
+		 * The other way is to fail to abort when encountering the second text node
+		 * because the kind of active format reconstruction isn't supported.
+		 */
+
+		if ( $processor->next_token() ) {
+			$this->assertSame(
+				array( 'HTML', 'BODY', 'P', 'B', '#text' ),
+				$processor->get_breadcrumbs(),
+				'Should have reconstructed the implicitly-closed B element.'
+			);
+		} else {
+			$this->assertSame(
+				WP_HTML_Processor::ERROR_UNSUPPORTED,
+				$processor->get_last_error(),
+				'Should have aborted for incomplete active format reconstruction when encountering the second text node.'
+			);
+		}
```

Comment:
So is the processor here not deterministic? Is `$processor->next_token()` not always returning the same value?

Reply:
I wrote this test with dmsnell/wordpress-develop#13 in mind. hopefully soon after 6.5 is out we'll have active format reconstruction working, and I didn't want this test to be misdirected when that starts up.

currently it aborts when encountering the need to reconstruct, meaning we can't verify the breadcrumbs, but simply detecting the aborted operation ensures we're doing the right thing from this point of view.

I've added a comment in 2e04332dd3. thanks for the question!